src/Microsoft.Extensions.Options/ConfigureOptions.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.Extensions.Options
 {
-    public class ConfigureOptions<TOptions> : IConfigureOptions<TOptions> where TOptions : class
+    public class ConfigureOptions<TOptions> : IConfigureOptions<TOptions>, IOptionsConfiguration<TOptions> where TOptions : class
```

Comment:
We should revisit the name of this class based on what we decide for the interface.